### DISCUSSION TITLE
Programmatically set multiple cursors

### DISCUSSION THREAD

**Discussion by squarewave17** [rennzhang/codemirror-editor-vue3#22] [General]
Hello! I'm trying to set a cursor at the beginning of each line, like you would with alt + click. There doesn't seem to be a way to do it with setCursor(). Does anyone know of a way to do this?

**Comment by rennzhang** [comment]
You can try the following methods:
```js
for (let i = 1, len = editor?.lineCount(); i < len; i++) {
  editor.addSelection({ line: i, ch: 0 });
}
editor.focus();

```

**Reply by squarewave17** [reply]
That worked perfectly! Thank you!
